rules/manager.go

```diff
@@ -506,38 +499,59 @@ func (m *Manager) ApplyConfig(conf *config.Config) error {
 // As there's currently no group syntax a single group named "default" containing
 // all rules will be returned.
 func (m *Manager) loadGroups(interval time.Duration, filenames ...string) (map[string]*Group, error) {
-	rules := []Rule{}
+	groups := make(map[string]*Group)
+
 	for _, fn := range filenames {
-		content, err := ioutil.ReadFile(fn)
-		if err != nil {
-			return nil, err
-		}
-		stmts, err := promql.ParseStmts(string(content))
-		if err != nil {
-			return nil, fmt.Errorf("error parsing %s: %s", fn, err)
+		rgs, errs := rulefmt.ParseFile(fn)
+		if errs != nil {
+			return nil, tsdb.MultiError(errs)
 		}
 
-		for _, stmt := range stmts {
-			var rule Rule
+		for _, rg := range rgs.Groups {
+			itv := interval
+			if rg.Interval != "" {
+				dur, err := model.ParseDuration(rg.Interval)
+				if err != nil {
+					return nil, err
+				}
+
+				itv = time.Duration(dur)
+			}
 
-			switch r := stmt.(type) {
-			case *promql.AlertStmt:
-				rule = NewAlertingRule(r.Name, r.Expr, r.Duration, r.Labels, r.Annotations)
+			rules := make([]Rule, 0, len(rg.Rules))
+			for _, r := range rg.Rules {
+				expr, err := promql.ParseExpr(r.Expr)
+				if err != nil {
+					return nil, err
+				}
 
-			case *promql.RecordStmt:
-				rule = NewRecordingRule(r.Name, r.Expr, r.Labels)
+				if r.Alert != "" {
+					fordur, err := model.ParseDuration(r.For)
+					if err != nil {
+						return nil, err
+					}
 
-			default:
-				panic("retrieval.Manager.LoadRuleFiles: unknown statement type")
+					rules = append(rules, NewAlertingRule(
+						r.Alert,
+						expr,
+						time.Duration(fordur),
+						labels.FromMap(r.Labels),
+						labels.FromMap(r.Annotations),
+					))
+					continue
+				}
+				rules = append(rules, NewRecordingRule(
+					r.Record,
+					expr,
+					labels.FromMap(r.Labels),
+				))
 			}
-			rules = append(rules, rule)
+
+			// Groups need not be unique across filenames.
```

Comment:
What do you mean labels?

I haven't yet touched the UI. But it would be simple to add the grouping there as I now added the filename also to the `Group` object. Currently in the UI, all the rules will be listed without any reference to the group/file.